frontend/src/Pages/FacilityManagement.tsx

```diff
@@ -40,73 +46,104 @@ export default function FacilityManagement() {
+            type: FormInputTypes.Dropdown,
+            label: 'Timezone',
+            interfaceRef: 'timezone',
+            required: true,
+            enumType: Timezones
         }
-        revalidate();
-        editFacilityModal.current?.close();
-        addFacilityModal.current?.close();
-        resetModal();
-    }
+    ];
 
-    const handleDeleteFacilityCancel = () => {
-        deleteFacilityModal.current?.close();
-        resetModal();
+    const addFacility: SubmitHandler<FieldValues> = async (data) => {
+        const response = await API.post('facilities', data);
+        if (!response.success) {
+            toaster('Failed to add facility', ToastState.error);
+            return;
+        }
+        toaster('Facility created successfully', ToastState.success);
+        void mutate();
+        addFacilityModal.current?.close();
+        setTargetFacility(null);
     };
-    const openDeleteFacility = (facility: Facility) => {
-        deleteFacilityModal.current?.showModal();
-        setTargetFacility(facility);
+
+    const updateFacility: SubmitHandler<FieldValues> = async (data) => {
+        const response = await API.patch(
+            `facilities/${targetFacility?.facility.id}`,
+            data
+        );
+        if (!response.success) {
+            toaster('Failed to update facility', ToastState.error);
+            return;
+        }
+        toaster('Facility updated successfully', ToastState.success);
+        void mutate();
+        editFacilityModal.current?.close();
+        setTargetFacility(null);
     };
-    const handleDeleteFacility = async () => {
-        if (targetFacility?.id == 1) {
+
+    const deleteFacility = async () => {
+        if (targetFacility?.facility.id == 1) {
             toaster('Cannot delete default facility', ToastState.error);
+            deleteFacilityModal.current?.close();
             return;
         }
-
-        await API.delete('facilities/' + targetFacility?.id)
-            .then((response) => {
-                if (response.success) {
-                    toaster(
-                        'Facility successfully deleted',
-                        ToastState.success
-                    );
-                    revalidate();
-                } else {
-                    toaster('Error deleting facility', ToastState.success);
-                }
-            })
-            .catch(() => {
-                toaster('Error deleting facility', ToastState.error);
-            });
-
+        const response = await API.delete(
+            'facilities/' + targetFacility?.facility.id
+        );
+        if (response.success) {
+            toaster('Facility successfully deleted', ToastState.success);
+            void mutate();
+        } else {
+            toaster('Error deleting facility', ToastState.error);
+        }
         deleteFacilityModal.current?.close();
-        resetModal();
-        void mutate();
+        setTargetFacility(null);
         return;
     };
```

Comment:
What do you think about something like putting the form inputs and the handlers into their own file so that we have a reusable form module for managing facilities? 

It cleans up the FacilityManagement page a bit more and adds an element of reusability (not that we have a use case for it yet).

I could be overthinking it and I am open to you pushing back on this.